utils/db/repos.ts

```diff
@@ -3,16 +3,33 @@ import type { DbRepo, RepoIdentifier } from '../../types/repository';
 import { convert } from './converter';
 
 export const getRepos = async (allRepos: RepoIdentifier[]) => {
-	const allReposFormattedAsTuples = allRepos.map(repo => `(${convert(repo.repo_provider)}, ${convert(repo.repo_owner)}, ${convert(repo.repo_name)})`).join(',');
-	const repo_list_q = `SELECT *
+	const batchSize = 50;
+	const allDbReposPromises = [];
+	for (let index = 0; index < allRepos.length; index += batchSize) {
+		const allReposSubset = allRepos.slice(index, index + batchSize);
+		const allReposFormattedAsTuples = allReposSubset.map(repo => `(${convert(repo.repo_provider)}, ${convert(repo.repo_owner)}, ${convert(repo.repo_name)})`).join(',');
+		const repo_list_q = `SELECT *
 		FROM repos 
 		WHERE (repo_provider, repo_owner, repo_name) IN (${allReposFormattedAsTuples})
 		ORDER BY repo_provider, repo_owner, repo_name`;
-	const result: { rows: DbRepo[] } = await conn.query(repo_list_q).catch(err => {
-		console.error(`[getRepos] Error in getting repository-list from the database`, { pg_query: repo_list_q }, err);
-		throw new Error(err); // FIXME: handle this more elegantly
-	});
-	return result.rows
+		const DbRepoSubsetPromise: Promise<{ rows: DbRepo[] }> = conn.query(repo_list_q).catch(err => {
+			console.error(`[getRepos] Error in getting repository-list from the database`, { pg_query: repo_list_q }, err);
+			throw new Error(`Error in getting repository-list from the database. Batch: ${index}:${index + batchSize - 1}. Error: ${err.message}`);
+		});
+		allDbReposPromises.push(DbRepoSubsetPromise);
+	}
+
+	const allDbRepos = await Promise.allSettled(allDbReposPromises).then(results => {
+		const allDbRepoLists = results.map((result) => {
+			if (result.status !== 'fulfilled') {
+				return [];
+			}
+			return result.value.rows;
+		})
+		return allDbRepoLists.flat();
+	})
+
+	return allDbRepos;
 }
```

Comment:
@coderabbitai, I don't think this is accurate. Since I have not awaited the async function call, the `throw` will not actually stop the execution for all the other batches. Have I gotten this wrong? Please let me know.